array.c

```diff
@@ -3521,6 +3521,120 @@ sort_2(const void *ap, const void *bp, void *dummy)
     return n;
 }
 
+/* Radix_Sort */
+
+typedef struct rsort_double_t { uint64_t z; VALUE v; } rsort_double_t;
+
+static int rsort_double(VALUE *const p, const long l) {
+
+    rsort_double_t *const _r = malloc(l * 2 * sizeof(rsort_double_t)),
+                          *a = _r, *b = a + l, *t;
+    if (_r == NULL) { return 1; } _Bool is_unordered = 0;
+
+    uint64_t F[8][256] = {{0}}, prev = 0;
+
+    for (long i = 0; i < l; i++) {
+        union { double d; uint64_t z; } u;
+        if (!RB_FLOAT_TYPE_P(p[i]) || isnan(u.d = rb_float_value(p[i]))) {
+            free(_r);
+            return 1;
+        }
+        u.z ^= ((int64_t)(u.z) >> 63) | ((uint64_t)1 << 63);
```

Comment:
This `63` is `SIZEOF_VALUE * CHAR_BIT -1`?
In other words, XOR-ing `u.z` and its 1bit right rotation?